app/controllers/posts_controller.rb

```diff
@@ -162,14 +162,18 @@ def delete_history
     audit_ids = @post.associated_audits.where(action: 'destroy').where(auditable_type: 'Reply') # all destroyed replies
     audit_ids = audit_ids.joins('LEFT JOIN replies ON replies.id = audits.auditable_id').where('replies.id IS NULL') # not restored
     audit_ids = audit_ids.group(:auditable_id).pluck(Arel.sql('MAX(audits.id)')) # only most recent per reply
-    @audits = Audited::Audit.where(id: audit_ids).paginate(per_page: 1, page: page)
+    @deleted_audits = Audited::Audit.where(id: audit_ids).paginate(per_page: 1, page: page)
 
-    if @audits.present?
-      @audit = @audits.first
+    if @deleted_audits.present?
+      @audit = @deleted_audits.first
       @deleted = Reply.new(@audit.audited_changes)
       @preceding = @post.replies.where('id < ?', @audit.auditable_id).order(id: :desc).limit(2).reverse
       @preceding = [@post] unless @preceding.present?
       @following = @post.replies.where('id > ?', @audit.auditable_id).order(id: :asc).limit(2)
+      @audits = @post.associated_audits.where(auditable_id: @following.map(&:id))
+      @audits = @audits.or(@post.associated_audits.where(auditable_id: @preceding.map(&:id))) unless @preceding.first.is_a?(Post)
```

Comment:
Can we get an explain on doing this with an or instead of just calculating the auditable_id first and then passing it to a single queryset? Otherwise this looks fine.

Reply:
good... question? the preceding code was slightly terrifying, I guess? xD that should work fine

Reply:
Better belated plan: Let's just set this to a `{}` and accept that the deleted replies page will not show history links? They seem unnecessary honestly?